test/AutoRest.TestServer.Tests/model-flattening.cs

```diff
@@ -0,0 +1,269 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using AutoRest.TestServer.Tests.Infrastructure;
+using model_flattening;
+using model_flattening.Models;
+using NUnit.Framework;
+
+namespace AutoRest.TestServer.Tests
+{
+    public class ModelFlatteningTests : TestServerTestBase
+    {
+        public ModelFlatteningTests(TestServerVersion version) : base(version, "model-flatten") { }
+
+        [Test]
+        public Task GetModelFlattenArray() => Test(async (host, pipeline) =>
+        {
+            var result = await new AllOperations(ClientDiagnostics, pipeline, host).GetArrayAsync();
+            Assert.AreEqual("1", result.Value.ElementAt(0).Id);
```

Comment:
Extract variables for commonly used expression like `result.Value.ElementAt(0)`

Reply:
Done.